lib/daru/io/importers/excel.rb

```diff
@@ -4,19 +4,29 @@ module Daru
   module IO
     module Importers
       class Excel < Base
-        Daru::DataFrame.register_io_module :from_excel, self
+        Daru::DataFrame.register_io_module :from_excel do |*args|
+          if args.first.end_with? '.xlsx'
+            require 'daru/io/importers/excelx'
```

Comment:
> That's inefficient. 

Not very. `require` works only once, the next time Ruby's enternal mechanizms will just check whether require'd file is loaded already.